src/charm.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm for GitHub Actions Exporter on kubernetes."""
+
+import logging
+from re import findall
+from typing import Dict
+
+from charms.nginx_ingress_integrator.v0.nginx_route import require_nginx_route
+from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
+from ops.charm import CharmBase, HookEvent, WorkloadEvent
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+
+logger = logging.getLogger(__name__)
+
+GH_EXPORTER_METRICS_PORT = 9101
+GH_EXPORTER_WEBHOOK_PORT = 8065
+SVC_HOSTNAME = "service-hostname"
+SVC_NAME = "service-name"
+SVC_PORT = "service-port"
+
+
+class GithubActionsExporterOperatorCharm(CharmBase):
```

Comment:
Please review the managing charm complexity spec and ensure that this charm is written in accordance of that spec

Reply:
- Charm Runtime State Abstraction - Moved all config to State class
- Separating Operational and Charm Logic - Moved get_exporter_version to GitHubActionsExporter class
- Charm Dependent Services - Moved ingress to ingress.py
- Using Shared Volumes - does not apply to this charm.
- Event Handler Names - OK

I was unsure about configuration validation. Should be moved too?